src/main/java/net/minecraftforge/client/model/ItemLayerModel.java

```diff
@@ -149,25 +164,39 @@ public ItemLayerModel read(JsonObject jsonObject, JsonDeserializationContext des
                 }
             }
 
-            var emissiveLayers = new IntOpenHashSet();
-            readUnlit(jsonObject, "emissive_layers", renderTypeNames, emissiveLayers, false);
-            boolean logWarning = readUnlit(jsonObject, "fullbright_layers", renderTypeNames, emissiveLayers, true); // TODO: Deprecated name. To be removed in 1.20
+            var emissiveLayers = new Int2ObjectArrayMap<ForgeFaceData>();
+            readUnlit(jsonObject, "forge_data", renderTypeNames, emissiveLayers, false);
+            boolean logWarning = readUnlit(jsonObject, "emissive_layers", renderTypeNames, emissiveLayers, true); // TODO: Deprecated name. To be removed in 1.20
+            logWarning |= readUnlit(jsonObject, "fullbright_layers", renderTypeNames, emissiveLayers, true); // TODO: Deprecated name. To be removed in 1.20
 
             return new ItemLayerModel(null, emissiveLayers, renderTypeNames, deprecated, logWarning);
         }
 
-        private boolean readUnlit(JsonObject jsonObject, String name, Int2ObjectOpenHashMap<ResourceLocation> renderTypeNames, IntOpenHashSet litLayers, boolean logWarning)
+        protected boolean readUnlit(JsonObject jsonObject, String name, Int2ObjectOpenHashMap<ResourceLocation> renderTypeNames, Int2ObjectMap<ForgeFaceData> layerData, boolean logWarning)
         {
             if (!jsonObject.has(name))
                 return false;
-            var fullbrightLayers = jsonObject.getAsJsonArray(name);
-            var renderType = new ResourceLocation("forge", "item_unlit");
-            for (var layer : fullbrightLayers)
+            JsonElement ele = jsonObject.get(name);
+            if(ele.isJsonArray()) // Legacy array-mode, all specified layers are max emissivity. TODO: To be removed in 1.20
             {
-                litLayers.add(layer.getAsInt());
-                renderTypeNames.putIfAbsent(layer.getAsInt(), renderType);
+                var fullbrightLayers = jsonObject.getAsJsonArray(name);
+                for (var layer : fullbrightLayers)
+                {
+                    layerData.put(layer.getAsInt(), new ForgeFaceData(0xFFFFFFFF, 15, 15));
+                }
+                return logWarning && !fullbrightLayers.isEmpty();
+            }
+            else // New mode, extra data is specified on a per-layer basis.
+            {
+                var fullbrightLayers = jsonObject.getAsJsonObject(name);
+                for (var layerStr : fullbrightLayers.keySet())
+                {
+                    int layer = Integer.parseInt(layerStr);
```

Comment:
Are `NumberFormatExceptions` handled here? I think they might need to be caught/wrapped into a `JsonParseException`

Reply:
They appear to be handled gracefully
`Unable to load model: 'minecraft:amethyst_shard#inventory' referenced from: minecraft:amethyst_shard#inventory: java.lang.NumberFormatException: For input string: "twebtey" `